.github/workflows/docker_build.yml

```diff
@@ -0,0 +1,68 @@
+name: Build Dockerfile
+on:
+  push:
+    branches: '*'
+jobs:
+  build_and_deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - uses: hadolint/hadolint-action@v1.6.0
+      with:
+        dockerfile: kuwala/core/cli/dockerfile
+    - uses: hadolint/hadolint-action@v1.6.0
+      with:
+        dockerfile: kuwala/core/jupyter/docker/dockerfile
+    - uses: hadolint/hadolint-action@v1.6.0
+      with:
+        dockerfile: kuwala/core/neo4j/docker/dockerfile
+    - uses: hadolint/hadolint-action@v1.6.0
+      with:
+        dockerfile: kuwala/core/neo4j/importer/dockerfile
+    - uses: hadolint/hadolint-action@v1.6.0
+      with:
+        dockerfile: kuwala/pipelines/google-poi/dockerfile
+    - uses: hadolint/hadolint-action@v1.6.0
+      with:
+        dockerfile: kuwala/pipelines/google-poi/src/pipeline/dockerfile
+    - uses: hadolint/hadolint-action@v1.6.0
+      with:
+        dockerfile: kuwala/pipelines/osm-poi/dockerfile
+    - uses: hadolint/hadolint-action@v1.6.0
+      with:
+        dockerfile: kuwala/pipelines/population-density/dockerfile
+
+    # This is the a separate action that sets up buildx runner
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1 
+    # So now you can use Actions' own caching!
+    - name: Cache Docker layers
+      uses: actions/cache@v2
+      with:
+        path: /tmp/.buildx-cache
+        key: ${{ runner.os }}-buildx-${{ github.sha }}
+        restore-keys: |
+          ${{ runner.os }}-buildx-
+
+    # And make it available for the builds
+    - name: Build neo4j and push
+      uses: docker/build-push-action@v2
+      with:
+        context: kuwala
+        file: kuwala/core/neo4j/docker/dockerfile
+        push: false
+        tags: kuwala/neo4j:${{ github.sha }}
+        cache-from: type=local,src=/tmp/.buildx-cache
+        cache-to: type=local,dest=/tmp/.buildx-cache-new
+      # This ugly bit is necessary if you don't want your cache to grow forever
+      # till it hits GitHub's limit of 5GB.
+      # Temp fix
+      # https://github.com/docker/build-push-action/issues/252
```

Comment:
I used this trick for caching in the local registry.
https://github.com/zero88/gh-registry/blob/main/README.md#usage

Reply:
good to know this tricks, thx~ 🎉

Reply:
oh, seems to me that they've fixed this issue ([PR](https://github.com/docker/build-push-action/issues/252#issuecomment-889326894))
I'll give this trick a shot next time (I need to change my cache ref, other would get this [error](https://github.com/david30907d/kuwala/runs/4556251620?check_suite_focus=true) 😅 )